src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int NOT_EXISTS = -1;
+    private static final int UNIT = 1;
+    private T[] internalArray;
+    private int capacity;
+    private int size;
+
+    public ArrayList() {
+        capacity = 10;
+        internalArray = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < capacity) {
+            internalArray[size] = value;
+            size++;
+        } else {
+            growIfArrayFull();
+            add(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == index) {
+            add(value);
+        } else if (isIndexAcceptable(index)) {
+            size++;
+            if (size == capacity) {
+                growIfArrayFull();
+            }
+            System.arraycopy(internalArray, index, internalArray,
+                    index + UNIT, size - index - UNIT);
+            internalArray[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Unavailable index "
+                    + "for adding element!!!");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (!isIndexAcceptable(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Can't get out of bound index!!!");
+        }
+        return internalArray[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (!isIndexAcceptable(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Can't get out of bound index!!!");
```

Comment:
It would be nice to show the index and current size of the array in the message.